src/lib/components/Molecule2dViewer.react.js

```diff
@@ -21,7 +21,23 @@ export default class Molecule2dViewer extends Component {
     }
 
     shouldComponentUpdate(nextProps) {
-        if (this.props.modelData !== nextProps.modelData) {
+        if (
+            this.props.modelData !== nextProps.modelData ||
+            (!this.props.selectedAtomIds && nextProps.selectedAtomIds) ||
+            (this.props.selectedAtomIds && !nextProps.selectedAtomIds) ||
+            (this.props.selectedAtomIds &&
+                nextProps.selectedAtomIds &&
+                this.props.selectedAtomIds.length !==
+                    nextProps.selectedAtomIds.length) ||
+            (this.props.selectedAtomIds &&
+                nextProps.selectedAtomIds &&
+                (this.props.selectedAtomIds.some(
+                    atomId => !(atomId in nextProps.selectedAtomIds)
+                ) ||
+                    nextProps.selectedAtomIds.some(
+                        atomId => !(atomId in this.props.selectedAtomIds)
+                    )))
+        ) {
```

Comment:
No obligation here but would suggest flattening this for readability.
Same comment applies to other `shouldComponentUpdate`.

```
shouldComponentUpdate(nextProps) {
    if (this.props.modelData !== nextProps.modelData) {
        return true;
    }

    // if one is defined but not the other (XOR)
    if (this.props.selectedAtomIds ? !nextProps.selectedAtomIds : nextProps.selectedAtomIds) {
        return true;
    }

    // etc.
}
```

Reply:
Opened an issue here: #419.